### ISSUE TITLE
Breaking change 13.0.0 => 13.1.0?

### ISSUE THREAD

**Issue by priecint** [FormidableLabs/redux-little-router#154]
Hey everyone  (and particulary @kkerr1),

seems like there was breaking change introduced with this PR https://github.com/FormidableLabs/redux-little-router/pull/150.

I followed [the guide](https://github.com/FormidableLabs/redux-little-router#redux-usage) to setup my regular webpage: menu with links on left and content on right.

There are two routes:
```javascript
const routes = {
    "/": {title: "Home"},
    "/merchants": {title: "Merchants"},
};

export const {
    reducer: routerReducer,
    middleware: routerMiddleware,
    enhancer: routerEnhancer,
} = routerForBrowser({ routes });
```
and for each route there is component to represent that page (so my intention is to display `HomePage` only for `/`, and `MerchantsPage` only for `/merchants`). This worked fine with 13.0.0 but with 13.1.0 HomePage never shows.
```jsx
<MainMenu>
    <Link href="/">
        <FormattedMessage
            id="mainMenu.items.home.label"
            defaultMessage="Home page"
        />
    </Link>
    <br/>
    <Link href="/merchants">
        <FormattedMessage
            id="mainMenu.items.merchants.label"
            defaultMessage="Merchants page"
        />
    </Link>
</MainMenu>

<Fragment forRoute="/">
    <MainContent>
        <Fragment forRoute="/merchants">
            <MerchantsPage />
        </Fragment>
        <Fragment forRoute="/">
            <HomePage />
        </Fragment>
    </MainContent>
</Fragment>
```
I created a [test case](https://github.com/priecint/redux-little-router/commit/147501eca980658df13866d68957e12e82ea601d) which I think should pass - in 13.0.0 it does, but in 13.1.0 it doesn't.

I tried changing order of the nested `Fragment`s (did not have any visible impact) and removing the top-most `Fragment` (`HomePage` was visible also for `/merchants`) but it didn't solve anything.

I'm going to use 13.0.0 for now but I think this is bug. What do you guys think? Thanks

**Comment by kkerr1** [comment]
Thanks @priecint this looks like the same issue as #153. I will take a look soon.

**Comment by Wolfhergang** [comment]
"exact" attribute feature is coming, i can feel it in my belly.

**Comment by kkerr1** [comment]
Should we expect `<HomePage/>` to be rendered only when the route is `/` or on any route? I am guessing we want it to only render on `/`? Maybe we do need to consider something like an `exact` prop, since we can fix this edge case by implementing the same behavior implemented in a894e0feb9be4ac457fda09c0f2b20ebef96f0f7 but perhaps that is a sort of strange API?

**Comment by priecint** [comment]
> this looks like the same issue as #153

It looks little bit different to me because @benmcmaster doesn't have the wrapping `<Fragment forRoute='/' />` as I do (notice I have two `<Fragment forRoute="/">` but he only has one). But I don't understand RLR very well so I'm not sure whether it makes difference or not.

> Should we expect `<HomePage/>` to be rendered only when the route is `/` or on any route?

`<HomePage/>` should be rendered only for `/`.

> Maybe we do need to consider something like an exact prop, since we can fix this edge case by implementing the same behavior implemented in a894e0f but perhaps that is a sort of strange API?

If I understand correctly, in 13.0.0 there was no need for `exact` so I don't think RLR needs it unless there is a technical reason for it. In 13.0.0 it was enough to specifiy `forRoute="/"` to say: "I want this to be the exact match." (Or at least I read it that way and it seemed comprehensible.)

**Comment by benmcmaster** [comment]
Hey @priecint, thank you for mentioning me here. It seems like our issues may be related. I just recreated your issue by nesting my routes within the `<Fragment forRoute="/">`. I am not seeing the Home element at all now:
```
const Demo = ({ router }) => {
  return (
    <div className={styles.container}>
      <div>
        <h1 className={styles.tagline}>
          <span className={styles.secondary}>A Compendium of</span>
          <br />
          <span className={styles.primary}>Ipsums and GIFs</span>
        </h1>

        <div className={styles.nav}>
          <Link href='/'>Home</Link>
          <Link href={{ pathname: '/cheese', query: { is: 'cheese' } }}>
            Cheese
          </Link>
          <Link href='/dog'>Dog</Link>
          <Link href='/cat?is=cat'>Cat</Link>
          <Link href='/hipster'>Hipster</Link>
        </div>

        <div className={styles.panes}>
          <Fragment key='/' forRoute='/'>
            <div>
              <Fragment key='/cheese' forRoute='/cheese'>
                <div>
                  <p>Cheese</p>
                </div>
              </Fragment>
              <Fragment key='/dog' forRoute='/dog'>
                <div>
                  <p>Dog</p>
                </div>
              </Fragment>
              <Fragment key='/cat' forRoute='/cat'>
                <div>
                  <p>Cat</p>
                </div>
              </Fragment>
              <Fragment key='/hipster' forRoute='/hipster'>
                <div>
                  <p>Hipster</p>
                </div>
              </Fragment>
              <Fragment key='/' forRoute='/'>
                <div>
                  <p>Home</p>
                </div>
              </Fragment>
            </div>
          </Fragment>
        </div>
      </div>
    </div>
  );
};
```
How is this supposed to work?... Seems like a very fundamental issue, unless I am missing something big... I am new to this plugin, but it seems like previous versions of this plugin worked with the `Fragement` configuration shown in [#153](https://github.com/FormidableLabs/redux-little-router/issues/153).

**Comment by kkerr1** [comment]
> this looks like the same issue as #153

I actually retract this comment. #153 is indeed somewhat different. Apologies for the confusion.

13.1.0 added a new behavior that says "When a `<Fragment forRoute="/">` route is relative (it is nested under a parent `<Fragment>` then treat that child fragment in such a way that it must exactly match the route of the parent `<Fragment>`. The parent fragment will continue to be a greedy match.  There looks to be at least edge-case bug where this behavior does not apply to a route where a `<Fragment forRoute="/">` route was nested  under a parent <Fragment forRoute="/">. 

In 13.0.0 it looks like if the `/` route is the last route, it will work correctly. But, I _think_ this was more of a happy coincidence than anything. Ideally, it should work the same regardless of order. I've got test cases for both this issue and #153 now, will hopefully steal enough time soon to work out a better solution soon.

**Comment by priecint** [comment]
@benmcmaster With regards to the latest comment seems like this is bug. Solution for me for now is to use the older version (13.0.0).

@kkerr1 Looking forward to testing this. Let me know if I can help somehow.

**Comment by MindFreeze** [comment]
I have also been relying on the behavior from 13.0.0 for a few routes and minor version bump with no warning of breaking changes really screwed me.
Will also revert to 13.0.0 for now until this is fixed.

Because I can't specify an array of routes anymore I also have this:
```
<Fragment
    forRoute="/"
    withConditions={location => ['/users', '/groups'].indexOf(location.route) !== -1}
>
```
which doesn't work in 13.1.0
Can't we just skip `forRoute` and rely solely on `withConditions` when it is specified? This way all cases can be easily covered by the user.

**Comment by tptee** [comment]
@MindFreeze this is definitely a bug, not an intentional breaking change. We'll have it sorted out soon

**Comment by tptee** [comment]
Published in `v13.1.2`
